src/plugins/template/backend/ops/string_tensor_unpack.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/reference/string_tensor_unpack.hpp"
+
+#include "evaluate_node.hpp"
+#include "string_tensor_unpack_shape_inference.hpp"
+
+template <>
+bool evaluate_node<ov::op::v15::StringTensorUnpack>(std::shared_ptr<ov::Node> node,
+                                                    ov::TensorVector& outputs,
+                                                    const ov::TensorVector& inputs) {
+    if (node->get_input_element_type(0) == ov::element::string) {
+        auto string_tensor_unpack = std::dynamic_pointer_cast<ov::op::v15::StringTensorUnpack>(node);
+        OPENVINO_ASSERT(string_tensor_unpack, "Node passed to StringTensorUnpack evaluate function is invalid.");
+        std::vector<ov::PartialShape> output_shapes;
+        output_shapes = ov::op::v15::shape_infer(string_tensor_unpack.get(),
+                                                 ov::util::get_tensors_partial_shapes(inputs),
+                                                 make_tensor_accessor(inputs));
+        auto outputs_it = outputs.begin();
+        for (const auto& p_shape : output_shapes) {
+            outputs_it->set_shape(p_shape.get_shape());
+            ++outputs_it;
+        }
+        const auto& data_shape = node->get_input_shape(0);
+        int64_t element_count = std::accumulate(data_shape.begin(), data_shape.end(), 1, std::multiplies<int64_t>());
+        ov::reference::string_tensor_unpack(inputs[0].data<const std::string>(),
+                                            outputs[0].data<int32_t>(),
+                                            outputs[1].data<int32_t>(),
+                                            outputs[2].data<uint8_t>(),
+                                            element_count);
```

Comment:
The `shape_size` helper will result with the same value but of `size_t` type instead `int64`. 
Maybe the string_count and element_count should be `size_t` as well?
```suggestion
        const auto element_count = shape_size(data_shape);
        ov::reference::string_tensor_unpack(inputs[0].data<const std::string>(),
                                            outputs[0].data<int32_t>(),
                                            outputs[1].data<int32_t>(),
                                            outputs[2].data<uint8_t>(),
                                            element_count);
```

Reply:
Done, type also changed to `size_t`.